Docs for NextRetryDelay (#2940)
temporalio/documentation: docs/encyclopedia/retry-policies.mdx

<PrettyImage src="/img/retry-interval-diagram.png" title="Diagram that shows the retry interval and its formula" />

### Per-error next retry delay

Sometimes, your Activity or Workflow raises a special exception that needs a different retry interval from the Retry Policy.

> To accomplish this, you may throw an [Application Failure](/references/failures#application-failure) with the next retry delay field set. This value will replace and override whatever the retry interval would be on the retry policy.

Reviewer: ```suggestion
To accomplish this, you may throw an [Application Failure](/references/failures#application-failure) with the Next Retry Delay field set. This value will replace and override whatever the retry interval would be on the Retry Policy.
```

Unclear what Next Retry Delay field is - is that a per-SDK field? Should it be capitalized?

Reviewer: Should this link right to the Java section that explains this?

> Note that your retries will still cap out under the retry policy's Maximum Attempts, as well as overall timeouts--i.e. if it's an activity, its Schedule-to-Close Timeout or, if it's a workflow, the Execution Timeout.

Reviewer: ```suggestion
Note that your retries will still cap out under the Retry Policy's Maximum Attempts, as well as overall timeouts. For an Activity, its Schedule-to-Close Timeout applies. For a Workflow, the Execution Timeout applies.
```

Reviewer: Rewrote for clarity- we try to avoid e.g. and i.e.
